File: src/main/java/org/usfirst/irs1318/gamesim/engine/actors/Robot.java
package org.usfirst.irs1318.gamesim.engine.actors;

import org.usfirst.irs1318.gamesim.engine.container.Container;

import java.util.Map;

public final class Robot extends Actor {
    public Robot(Location location, Map<String, Container> containers) {
        super(location, containers);
    }
}

File: src/main/scala/org/usfirst/irs1318/gamesim/engine/actors/Robot.scala
package org.usfirst.irs1318.gamesim.engine.actors

import org.usfirst.irs1318.gamesim.engine.container.Container

case class Robot(name: String, location: Location, containers: Map[String, Container]) extends Actor {
  override def mapLocation(fn: Location => Location): Actor = copy(location = fn(location))
}
